### DISCUSSION TITLE
A new way to think about the transit tiles and multimodal routing

### DISCUSSION THREAD

**Discussion by pranavpandey1998official** [valhalla/valhalla#2877] [Ideas]
I don't think I completely understand the whole flow or code but I love to share my thought and maybe someone from the community can correct me, or say why they have taken a different approach to the problem,

So lets start at a very higher level when I understand a vehicle/ person moving form one place to another I can see that first he has to take a local road from which connects him to a main road that inturn can connect to him to a main highway , our cities are designed that way, and this is  called `HUB and Spoke model`, 

Valhalla is based on this intuition and build on top of it , 
Now when it comes to transit I can Imagine two type of hierarchy that a person can take either he can walk or he can take the transit, and in multi-modal based on these two hierarchies we should do routing, So intuitively I can imagine two hierarchies one road network and one transit network ,

Now looking at the code I can see that the transit hierarchies is embedded with the road network tile and separate data is stored like stops , ingress and platform to switch between the two hierarchies (road and transit) , 

so my question is why didn't the community choose to build a hierarchy tile specially for transit,
then we will have 4 hierarchy -> transit, highway, arterial, and local
 isn't it more intuitive and also works perfectly with the current implementation , we don't have to maintain the stops , ingress and platform but nodeTransitions which will connect the road hierarchy tile to the transit hierarchy tile, 

for transit tile we can extend the base tile and add extra information that transit need and can extend nodeTransition to store the extra info we need

Can someone from the community explain why they didn't take this approach and what are the complication you can think of if you have taken this approach

ps with this we can support extended GTFS like continuous pickup and dropoff

**Comment by dnesbitt61** [comment]
Transit data is a 4th hierarchy level separate from the 3 road hierarchies (local, arterial, highway). However, it does not use NodeTransitions (they were developed after the initial transit support). Node transitions only happen at nodes (intersections) that contain roads on multiple hierarchies at the same node location (lat,lng). Since transit is imported separately from OSM (via GTFS) it must be connected to the existing road network in some manner. That is what the transit "connection" edges are for - they connect transit stops (which are not colocated with nodes in the road network) to nodes in the road network. These connection edges provide the geometry from the node to the transit stop. Without these edges, transit stops would have to be at the same location as the node on the road network.

**Comment by pranavpandey1998official** [comment]
hi @dnesbitt61 , I understand that they are at different hierarchy, but what I want to know is that why is the data about the transit stored in every tile, like when we store tiles we have them named 1/345/234 here 1 means that that is arterial tile and also when graph reader reads it it treats it like that but for transit we don't have 4/233/323 instead in each hierarchy tile we store the information about the transit , instead of doing that if we were to make a separate tile completely for transit and connect it to the rest of the tiles like we connect local, hierarchy and highway doesn't that makes sense and also brings up more opportunity for expansion
